src/lib.rs

```diff
@@ -14,64 +16,71 @@ pub struct CInfo {
 }
 
 #[no_mangle]
-pub extern fn whatlang_detect(ptr: *const c_char, cinfo: &mut CInfo) -> u8 {
-    let cs = unsafe { CStr::from_ptr(ptr) };
+pub extern "C" fn whatlang_detectn(ptr: *const c_char, len: libc::size_t, cinfo: &mut CInfo) -> u8 {
+    let text = unsafe {
+        core::slice::from_raw_parts(ptr as *const u8, len)
+    };
+	detect_internal(&text, cinfo)
```

Comment:
There's a couple of places where the spacing looks off here. I'm not sure of this repo's policy for formatting, it looks like `cargo fmt` will touch a few other places than what you've changed, but I'd personally prefer sticking to cargo fmt's choices.

Reply:
Reformatted the whole thing.